### PR TITLE
Add Editor.IsTextPredictionEnabled property

### LINKED ISSUE #3555: [Enhancement] Editor: Control over text-prediction
## Rationale

#1677 for Editor

`Editor`s do not currently allow users to disable text prediction/autocomplete/autosuggest without a custom renderer or effect.

## Implementation

Add bindable property to `Editor`:

```csharp
public static readonly BindableProperty IsTextPredictionEnabledProperty = BindableProperty.Create("IsTextPredictionEnabled", typeof(bool), typeof(Editor), true);

```

Renderers will need to be updated accordingly.

### Android

- `IsTextPredictionEnabled == true`: Default. Text prediction is enabled.
- `IsTextPredictionEnabled == false`: Text prediction is disabled.

### iOS

- `IsTextPredictionEnabled == true`: Default. Text prediction is enabled. Note: may also disable autocorrect.
- `IsTextPredictionEnabled == false`: Text prediction is disabled.

### UWP

- `IsTextPredictionEnabled == true`: Default. Text prediction  is enabled.
- `IsTextPredictionEnabled == false`: Text prediction is disabled.

## Implications for CSS

None

## Backward Compatibility

Since we are adding new properties, there should be no compatibility problems as long as:

1. We ensure that an `Editor`'s text prediction defaults to enabled, as is the expected behavior now,
2. We ensure that any Effects or Custom Renderers that may have been created to produce this behavior take precedence over any values that we set, to the best of our ability.

Third party renderers will need to be updated to ensure that this functionality is officially supported.

## Difficulty : Easy-Medium

This is an easy-medium change and great for a first-time contributor. Biggest challenge will be ensuring that disabling text prediction does not also disable other features of the `Editor` view on all platforms.

### PR DESCRIPTION
**Pull request by noamyogev84**
Fixes #3555
### Description of Change ###
Added the `Editor.IsTextPredictionEnabledProperty` field and `Editor.IsTextPredictionEnabled` property. Modified `EditorRenderer.OnElementPropertyChanged()` for Android, iOS and UWP to react to changes to the new `IsTextPredictionEnabled` property.

On Android setting `IsTextPredictionEnabled to false sets the `InputTypes.TextFlagNoSuggestions` flag on the native control's `InputType`.

On UWP the native control's `IsTextPredictionEnabled` will be set to the value of the `Editor.IsTextPredictionEnabled` property.

On iOS the native control's `AutocorrectionType` property will be set to `No` when `Editor.IsTextPredictionEnabled` is `false`.

### Issues Resolved ### 

- Fixes #3555

### API Changes ###

- Core/XAML (all platforms)
- iOS
- Android
- UWP

### Behavioral/Visual Changes ###

Please note #2038

### Before/After Screenshots ### 

Not applicable

### PR Checklist ###

- [x] Has automated tests 
- [x] Rebased on top of the target branch at time of PR
- [x] Changes adhere to coding standard

### CODE REVIEW

File: Xamarin.Forms.Core/Editor.cs [hunk 1]
```diff
@@ -23,6 +23,8 @@ public class Editor : InputView, IEditorController, IFontElement, IPlaceholderEl
 
 		public static readonly BindableProperty PlaceholderColorProperty = PlaceholderElement.PlaceholderColorProperty;
 
+		public static readonly BindableProperty IsTextPredictionEnabledProperty = BindableProperty.Create(nameof(IsTextPredictionEnabled), typeof(bool), typeof(Editor), true, BindingMode.OneTime);
```
Comment (hartez) [inline Xamarin.Forms.Core/Editor.cs]: Why set the default binding mode to `OneTime` rather than `Default`?

File: Xamarin.Forms.Platform.Android/Renderers/EditorRenderer.cs [hunk 2]
```diff
@@ -165,12 +167,23 @@ void UpdateInputType()
 			var keyboard = model.Keyboard;
 
 			edit.InputType = keyboard.ToInputType() | InputTypes.TextFlagMultiLine;
-			if (!(keyboard is Internals.CustomKeyboard) && model.IsSet(InputView.IsSpellCheckEnabledProperty))
+			if (!(keyboard is Internals.CustomKeyboard))
 			{
-				if ((edit.InputType & InputTypes.TextFlagNoSuggestions) != InputTypes.TextFlagNoSuggestions)
+				if (model.IsSet(InputView.IsSpellCheckEnabledProperty))
```
Comment (hartez) [inline Xamarin.Forms.Platform.Android/Renderers/EditorRenderer.cs]: Can we simplify these next few `if` statements? I think the end result is that `TextFlagNoSuggestions` is always set to 'on' if `IsSpellCheckEnabled` or `IsTextPredictionEnabled` are true, and remains unchanged otherwise?

File: Xamarin.Forms.Core/Editor.cs (hunk 1 above)
Comment (noamyogev84) [reply Xamarin.Forms.Core/Editor.cs]: According to documentation, the default will fallback to one way. So editor will be monitored for value changes. Is this behavior OK?

File: Xamarin.Forms.Platform.Android/Renderers/EditorRenderer.cs (hunk 2 above)
Comment (noamyogev84) [reply Xamarin.Forms.Platform.Android/Renderers/EditorRenderer.cs]: looking into that.

File: Xamarin.Forms.Core/Editor.cs (hunk 1 above)
Comment (hartez) [reply Xamarin.Forms.Core/Editor.cs]: Yes. I think the least surprising thing for users will be for it to behave like the other BindableProperties by default.

**Comment by noamyogev84** [comment]
Hi @hartez .
Would you take pick at my changes? Thanks.

File: Xamarin.Forms.Core/Editor.cs (hunk 1 above)
Comment (noamyogev84) [reply Xamarin.Forms.Core/Editor.cs]: Hi @hartez .
Would you take pick at my changes? Thanks.

**Comment by noamyogev84** [comment]
@hartez @PureWeen Hey guys. Is there anything else that needs work on my side?

**Comment by PureWeen** [comment]
No apologies for the delays
Can you rebase this to fix the conflict and then we'll get it merged in? If you don't have time let me know and I can get it rebased

File: .Xamarin.forms.sln [hunk 3]
```diff
@@ -142,7 +142,6 @@ Global
 	GlobalSection(SharedMSBuildProjectFiles) = preSolution
 		Xamarin.Forms.Controls.Issues\Xamarin.Forms.Controls.Issues.Shared\Xamarin.Forms.Controls.Issues.Shared.projitems*{0a39a74b-6f7a-4d41-84f2-b0ccdce899df}*SharedItemsImports = 4
 		Xamarin.Forms.Core.UITests.Shared\Xamarin.Forms.Core.UITests.projitems*{0a39a74b-6f7a-4d41-84f2-b0ccdce899df}*SharedItemsImports = 4
-		Xamarin.Forms.Controls.Issues\Xamarin.Forms.Controls.Issues.Shared\Xamarin.Forms.Controls.Issues.Shared.projitems*{0f0db9cc-ea65-429c-9363-38624bf8f49c}*SharedItemsImports = 13
```
Comment (PureWeen) [inline .Xamarin.forms.sln]: this commit looks like it snuck in.
Can you undo the changes to this file?

Comment (noamyogev84) [reply .Xamarin.forms.sln]: @PureWeen done. 
Go ahead and take over if you see any problems.
